bpf/bpf_overlay.c

```diff
@@ -455,6 +455,9 @@ static __always_inline int handle_ipv4(struct __ctx_buff *ctx,
 	/* A packet entering the node from the tunnel and not going to a local
 	 * endpoint has to be going to the local host.
 	 */
+
+	set_identity_mark(ctx, *identity, MARK_MAGIC_IDENTITY);
+
```

Comment:
If this gets scrubbed during the veth traversal, then `to-netdev` would just not see the magic mark and keep the `src_sec_identity` as `UNKNOWN_ID`. I suppose that's alright :).

Reply:
I think so too